modules/apps/cookies/cookies-impl/src/main/java/com/liferay/cookies/internal/manager/CookiesManagerImpl.java

```diff
@@ -336,6 +337,10 @@ public void validateSupportCookie(HttpServletRequest httpServletRequest)
 	private Map<String, Cookie> _getCookiesMap(
 		HttpServletRequest httpServletRequest) {
 
+		if (httpServletRequest == null) {
+			return new HashMap<>();
+		}
+
```

Comment:
Prior to this change, this method would always return a mutable object which is backed by the request attribute and vica-versa. If that couldn't be returned, a NPE is thrown.

With the change there will be two `addCookie()` invocations (`TransferHeadersHelperImpl` and `Header`) that do not get added to the  `Map<String, Cookie>` (`cookieMap`) stored into the request attribute (with key resolved by `CookiesManagerImpl.class.getName()` ).

I would have to dig deeper if this is a significant problem, but maybe you have already done this?

Reply:
I am just a bit concerned that it allows for a silent failure. Did you consider putting the `if (httpServletRequest != null)`  into those other usage points instead of within `_getCookiesMap()` ? Like you did in `addCookie()`. That would be handling it at source IMHO :)